Sources/Factory/Factory/Registrations.swift

```diff
@@ -309,32 +309,44 @@ internal struct FactoryOptions {
 
 extension FactoryOptions {
     /// Internal function to return factory based on current context
-    func factoryForCurrentContext() -> AnyFactory?  {
+    func factoryForCurrentContext(resolveContext: FactoryContext?) -> AnyFactory?  {
         if let contexts = argumentContexts, !contexts.isEmpty {
-            for arg in FactoryContext.arguments {
-                if let found = contexts[arg] {
-                    return found
-                }
-            }
-            for (_, arg) in FactoryContext.runtimeArguments {
+            let arguments = resolveContext?.arguments ?? FactoryContext.arguments + FactoryContext.runtimeArguments.values
+
+            for arg in arguments {
                 if let found = contexts[arg] {
                     return found
                 }
             }
         }
+
         if let contexts = contexts, !contexts.isEmpty {
+            let isPreview: Bool
+            let isTest: Bool
+            let isSimulator: Bool
+
+            if let resolveContext {
+                isPreview = resolveContext.isPreview || FactoryContext.isPreview
```

Comment:
Not sure your "override" is doing what you think. If you're trying to force the context to be "simulator" but the FactoryContext.isTest result is true that condition will resolve before your forced comparison.

Reply:
I thanks might be something I have to fix as I first made it with the inclusion of the refactor #105 but that proved to be difficult to merge with develop. So I restarted with a more modest approach. Will look into this today.

Reply:
A solution would be to only allow to overwrite arguments and not the isTest or other build context related things.

I will ponder a bit on that and try to propose something ...